src/network/networkd-link.c

```diff
@@ -110,7 +110,7 @@ static bool link_ipv6_enabled(Link *link) {
         if (!socket_ipv6_is_supported())
                 return false;
 
-        return link_dhcp6_enabled(link) || link_ipv6ll_enabled(link) || network_has_static_ipv6_addresses(link->network);
+        return link_ipv6ll_enabled(link) || network_has_static_ipv6_addresses(link->network);
```

Comment:
Can you add a comment explaining that dhcp6 requires ipv6ll here? I think this will help future readers.